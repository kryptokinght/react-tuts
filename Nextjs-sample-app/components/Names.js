

class Names extends React.Component {
    state = {
        show: false
    }

    render() {
        let list = <p>Nothing to show</p>;

        if(this.state.show)
            list = populateList(this.props.names);

        return(
            <div>
                <ul className="list-group">
                    {list}
                </ul>
                <button className="btn btn-lg btn-success" onClick={e => this.setState({show:!this.state.show})}>
                    Show
                </button>
            </div>
        );
    }
}

function populateList(names) {
    let list = [];
    names.forEach((name) => {
        list.push(<li className="list-group-item" key={name}>{name}</li>) 
    });
    return list;
}

export default Names;