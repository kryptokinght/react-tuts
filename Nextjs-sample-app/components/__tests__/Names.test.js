import React from 'react';
import Names from '../Names';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'; //removes unnecessary JSON from snapshot

describe('<Names />', () => {
    it('should be defined', () => {
        expect(Names).toBeDefined();
    });
    it('should render', () => {
        const wrapper = shallow(<Names />);
        expect(shallowToJson(wrapper))
        .toMatchSnapshot();
    });
});