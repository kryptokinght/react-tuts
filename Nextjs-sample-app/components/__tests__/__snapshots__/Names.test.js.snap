// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Names /> should render 1`] = `
<div>
  <ul
    className="list-group"
  >
    <p>
      Nothing to show
    </p>
  </ul>
  <button
    className="btn btn-lg btn-success"
    onClick={[Function]}
  >
    Show
  </button>
</div>
`;
